game/src/core/components/DrawComponent.java

```diff
@@ -340,148 +319,38 @@ public void animationMap(final Map<String, Animation> animationMap) {
-            LinkedList<IPath> list = new LinkedList<>();
-            list.add(new SimpleIPath(fileName));
-            storage.put(lastDir, list);
-          }
-        }
-      }
-    }
-
-    // sort the files in lexicographic order (like the most os)
-    // animations will be played in order
-    storage.values().forEach(x -> x.sort(Comparator.comparing(IPath::pathString)));
-    // create animations
-    storage.forEach(
-        (name, textureSet) -> animationMap.put(name, Animation.fromCollection(textureSet)));
-    jar.close();
-  }
-
-  /**
-   * Load animations if the game is running in the IDE (or over the shell).
-   *
-   * @param path Path to the animations.
-   */
-  private void loadAnimationsFromIDE(final IPath path) {
-    URL url = DrawComponent.class.getResource("/" + path.pathString());
-    if (url != null) {
-      try {
-        try {
-          File apps = new File(url.toURI());
-          animationMap =
-              Arrays.stream(Objects.requireNonNull(apps.listFiles()))
-                  .filter(File::isDirectory)
-                  .collect(Collectors.toMap(File::getName, DrawComponent::allFilesFromDirectory));
-        } catch (IllegalArgumentException e) {
-          LOGGER.log(
-              CustomLogLevel.ERROR, "Could not load animations from directory: " + url.toURI(), e);
-        }
-      } catch (URISyntaxException e) {
-        LOGGER.log(CustomLogLevel.ERROR, "Could not load animations from directory", e);
-      }
+  // Helper method that loads the animation assets from the given path. Called by the constructor,
+  // the animation map will be constructed from the subdirectories of the given path.
+  private void loadAnimationAssets(final IPath path) {
+    final Map<String, List<IPath>> subdirectoryMap = new HashMap<>();
+    try {
+      FileSystemUtil.visitResources(
+          path.pathString(),
+          new SimpleFileVisitor<>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+              if (Files.isRegularFile(file)) {
+                subdirectoryMap
+                    .computeIfAbsent(
+                        file.getParent().getFileName().toString(), k -> new ArrayList<>())
+                    .add(new SimpleIPath(file.toString()));
+              }
+              return FileVisitResult.CONTINUE;
+            }
+          });
+    } catch (Exception e) {
+      throw new RuntimeException("File not found: " + path, e);
     }
+    // A Map with sorted values (IPath lists) in natural string order (ascending)
+    animationMap =
+        subdirectoryMap.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    x ->
+                        Animation.fromCollection(
+                            x.getValue().stream()
+                                .sorted(Comparator.comparing(IPath::pathString))
+                                .toList())));
```

Comment:
Das ist super schlecht lesbar - Du machst einen Stream über einer Menge auf und macht sofort einen komplexen Abschluss per Collect. In meinem Buch ist sowas ein Code Smell. Bitte lesbar gestalten.